### PR TITLE
Update to 3.0.0rc4

### PR DESCRIPTION
**Pull request by jtpio**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jtpio** [comment]
This would also require an rc release of `jupyterlab_server`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,12 +22,12 @@ build:
...
-    - python >=3.5
+    - python >=3.6
     - pip
     - nodejs >=10
   run:
-    - python >=3.5
-    - jupyterlab_server >=1.0.0,<2.0.0
+    - python >=3.6
+    - jupyterlab_server >=2.0.0rc1,<3.0.0
     - notebook >=4.3.1
```
Comment (jasongrout) [inline recipe/meta.yaml:31]: Are we requiring notebook, or can this be removed?

Comment (jtpio) [reply recipe/meta.yaml:31]: Good point. It should be possible to update the requirements according to: https://github.com/jupyterlab/jupyterlab/blob/2bade97edadea865b294eb49c8d693d4261ca404/setup.py#L144-L152

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,13 +22,14 @@ build:
...
     - pip
     - nodejs >=10
   run:
-    - python >=3.5
-    - jupyterlab_server >=1.0.0,<2.0.0
-    - notebook >=4.3.1
+    - python >=3.6
+    - jupyterlab_server >=2.0.0rc1,<3.0.0
+    - jupyter_server >=1.0.1,<2.0.0
+    - nbclassic >=0.2.0,<1.0.0
```
Comment (jasongrout) [inline recipe/meta.yaml:32]: according to semver, 0.3 could have breaking changes from 0.2. Not sure what we are following with nbclassic, though.

(well, technically any pre-1.0 release can do whatever it wants, but the typical ^ implies breaking changes possible between 0.2 and 0.3

All this to say that either nbclassic should probably either bump to 1.0, or we should restrict to 0.2.0 to `<0.3.0`.

Comment (jtpio) [reply recipe/meta.yaml:32]: Yes `nbclassic >=0.2.0,<0.3.0` sounds good for now. Unless it's ready for 1.0.

**Comment by jtpio** [comment]
TODO:

- [ ] rc release for `jupyterlab_server`
- [ ] package `nbclassic`

**Review by afshin — APPROVED** [review]
Cheers!
👍

**Comment by jtpio** [comment]
@jasongrout should the `jupyterlab_server_rc` from https://github.com/conda-forge/jupyterlab_server-feedstock/pull/28 be specified in the config?

**Comment by jasongrout** [comment]
Yes, following the instructions at https://github.com/conda-forge/cfep/blob/master/cfep-05.md#specification

**Comment by jtpio** [comment]
Perfect, thanks

**Comment by jtpio** [comment]
@conda-forge-admin, please rerender

**Comment by jtpio** [comment]
Looks like the `staging` folder is missing in the `3.0.0rc2` tarball:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=216274&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=2223

https://pypi.org/project/jupyterlab/3.0.0rc2/#files


While `3.0.0b4` still has it: https://pypi.org/project/jupyterlab/3.0.0b4/#files

**Comment by jasongrout** [comment]
> Looks like the `staging` folder is missing in the `3.0.0rc2` tarball:

I checked back as far as rc0, I think, and didn't see the staging folder, in case that helps narrow things down.

**Comment by jtpio** [comment]
Yes `3.0.0b4` is the last release to have it.

Which is just before the switch to `jupyter_packaging`: https://github.com/jupyterlab/jupyterlab/blob/v3.0.0b4/setupbase.py

**Comment by jasongrout** [comment]
I just published a new release of jupyter-packaging, and used that to publish rc3, which now has the staging directory. Can you try updating to rc3 and see if that works?

**Comment by jasongrout** [comment]
Actually, that was easy enough to change - I just bumped to rc3 - hopefully that works!

**Comment by jtpio** [comment]
Nice, all green now with rc4 and the change from https://github.com/jupyterlab/jupyterlab/pull/9129.

**Comment by jasongrout** [comment]
Thanks @jtpio!

**Pull request merged**